stock_operating_unit/model/stock_picking.py

```diff
@@ -0,0 +1,56 @@
+# © 2019 ForgeFlow S.L.
+# © 2019 Serpent Consulting Services Pvt. Ltd.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    operating_unit_id = fields.Many2one(
+        "operating.unit",
+        "Requesting Operating Unit",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+
+    @api.onchange("picking_type_id", "partner_id")
+    def onchange_picking_type(self):
+        res = super(StockPicking, self).onchange_picking_type()
+        if self.picking_type_id:
+            unit = self.picking_type_id.warehouse_id.operating_unit_id
+            self.operating_unit_id = unit
+        return res
+
+    @api.constrains("operating_unit_id", "company_id")
+    def _check_company_operating_unit(self):
+        for rec in self:
+            if (
+                rec.company_id
+                and rec.operating_unit_id
+                and rec.company_id != rec.operating_unit_id.company_id
+            ):
+                raise UserError(
+                    _(
+                        "Configuration error. The Company in the Stock Picking "
+                        "and in the Operating Unit must be the same."
+                    )
+                )
+
+    @api.constrains("operating_unit_id", "picking_type_id")
+    def _check_picking_type_operating_unit(self):
+        for rec in self:
+            warehouse = rec.picking_type_id.warehouse_id
+            if (
+                rec.picking_type_id
+                and rec.operating_unit_id
```

Comment:
```suggestion
                and rec.operating_unit_id
                and warehouse.operating_unit_id
```

This function should add domain for case warehouse don't have operating unit.
In my case:
Warehouse OU : False
User OU: Main OU

Step to test:
1. Create warehouse without OU.
2. Create PO (Main OU) and select product type Storable Product
3. Click Confirm button
4. It throw error because warehouse ou != my ou

Is it can receipt stock to warehouse?

-----------------------------

Edited:
We can use Main Operating Unit for all user.

Reply:
Hello, I think this may seem a valid point. If we don't fix it, it means, we can't setup Warehouses with no operating unit (which is also legit to me in some use case, where the whole company may want to allow user to cross warehouse).

But I also think that, this warning may come with good reason.

Can some experts in this area clarify this.

@max3903 @JordiBForgeFlow @HviorForgeFlow

Reply:
@Saran440 let's merge this as is for now? I think most people can get along with this (me included).

Reply:
ping @AaronHForgeFlow aka OU master

Reply:
It seems this https://github.com/OCA/operating-unit/pull/234 is not included here. I think this would solve the issue. Can you check?

Reply:
Ok